manager2/src/app/admin/projects/projects.component.ts

```diff
@@ -170,40 +170,51 @@ export class ProjectsComponent implements OnInit {
             'orga': this.new_project.orga,
             'path': this.new_project.path,
-            'expire': new Date(this.new_project.expire).getTime()}
-                               ).subscribe(
-                                   resp => {
-                                       this.add_project_msg = resp.message;
-                                       this.project_list();
-                                       this.pending_list(true);
-                                       this.userService.addToProject(this.new_project.owner, this.new_project.id).subscribe(
-                                           resp => {
-                                               this.new_project = {};
-                                           },
-                                           err => {
-                                               console.log('failed  to add user to project');
-                                               this.add_project_error_msg = err.error.message;
-                                           }
-                                       )
-
-                                   },
-                                   err => {
-                                       console.log('failed to add project', this.new_project);
-                                       this.add_project_error_msg = err.error.message;
-                                   }
-                               );
+            'expire': new Date(this.new_project.expire).getTime()
+        }
+        ).subscribe(
+            resp => {
+                this.add_project_msg = resp.message;
+                this.project_list();
+                this.pending_list(true);
+                this.userService.addToProject(this.new_project.owner, this.new_project.id).subscribe(
+                    resp => {
+                        this.new_project = {};
+                    },
+                    err => {
+                        console.log('failed  to add user to project');
+                        this.add_project_error_msg = err.error.message;
+                    }
+                )
+
+            },
+            err => {
+                console.log('failed to add project', this.new_project);
+                this.add_project_error_msg = err.error.message;
+            }
+        );
     }
 
-    project_list(refresh_requests = false){
+    project_list(refresh_requests = false) {
         this.projects = [];
         this.projectService.list(true).subscribe(
             resp => {
-                if(resp.length == 0) {
+                if (resp.length == 0) {
                     return;
                 }
                 let data = resp;
-                for(var i=0;i<data.length;i++){
-                    data[i].expire = new Date(data[i].expire);
+                for (var i = 0; i < data.length; i++) {
+                    if (!data[i].expire) {
+                        data[i].expire = new Date(data[i].expire);
```

Comment:
? If not expire you set a new date with expire

Reply:
Yes sorry I was in a meeting when a wrote this code ;) 
I will fix it

Reply:
It is a copy/past error, the original commit with this condition is here : https://github.com/genouest/genouestaccountmanager/commit/43e9f04f6d837f5d4a1a209c4e5b41816a9a1690